src/pg_query_deparse.c

```diff
@@ -2276,7 +2276,12 @@ static void deparseRangeVar(StringInfo str, RangeVar *range_var, DeparseNodeCont
 
 static void deparseRawStmt(StringInfo str, RawStmt *raw_stmt)
 {
-	deparseStmt(str, raw_stmt->stmt);
+	if(raw_stmt->stmt != NULL) {
+		deparseStmt(str, raw_stmt->stmt);
+		return;
+	}
+
+	elog(ERROR, "deparse error in deparseRawStmt: RawStmt with empty Stmt");
```

Comment:
Whilst I generally like the idea of returning early, I think this makes it a bit confusing here - maybe we can write like this instead:

```c
if (raw_stmt->stmt == NULL)
  elog(ERROR, "deparse error in deparseRawStmt: RawStmt with empty Stmt");

deparseStmt(str, raw_stmt->stmt);
```

Note that `elog(ERROR, ...)` will raise an exception (or the C equivalent of it, i.e. unwind the stack), so you don't need to have a return in that if condition.

Reply:
I didn't know that, I just dabble in C. I will update the PR with your suggestion.